runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricUpdatesProtos.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.core.metrics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import org.apache.beam.model.fnexecution.v1.BeamFnApi.MonitoringInfo;
+import org.apache.beam.runners.core.construction.metrics.MetricKey;
+
+/** Convert {@link MetricUpdates} to a {@link List} of {@link MonitoringInfo}. */
+public class MetricUpdatesProtos {
+  public static List<MonitoringInfo> toProto(MetricUpdates metricUpdates) {
+    ArrayList<MonitoringInfo> monitoringInfos = new ArrayList<>();
+
```

Comment:
Please share code with 

https://github.com/apache/beam/blob/5b1700db834cf949547a20a30f3ee1270710585a/runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java#L147

Reply:
Slightly confused; you're pointing at `MetricsContainerImpl.counterUpdateToMonitoringInfo`, which converts a counter `MetricUpdate` to a `MonitoringInfo`.

In this PR I moved that logic here (a file more explicitly dedicated to MetricUpdate<->MI conversions), removed it there, and generalized it to other metric types. It is still only called from one place, but is reusable if other MetricUpdate<->MI code-paths crop up (it was previously a private method).

It also now uses MI-builder stuff for setting {name,labels} from a MetricKey ([`handleMetricKey` below](https://github.com/apache/beam/pull/7934/files#diff-4f9f0786ec97d7727d0d393e06b2d5f8R49)), so that code is now going through a more widely-shared path.

lmk if I'm missing something.